src/main/java/Duke.java

```diff
@@ -1,43 +1,96 @@
+import java.util.Locale;
 import java.util.Scanner;
 
 public class Duke {
 
+    private static final String SOMETHING_WRONG = "Sorry, something went wrong. ";
+
+    private static int nextAdd = 0;
+
     private static void firstGreet() {
         System.out.println("Hello! I'm Duke!");
         System.out.println("What can I do for you?");
         System.out.println("-------------------------------------");
     }
 
+    private static String createRemainingString(String[] inputWords) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 2; i < inputWords.length; i++) {
+            sb.append(inputWords[i]);
+            if (i < inputWords.length - 1) {
+                sb.append(" ");
+            }
+        }
+        return sb.toString();
+    }
+
+    private static void makeTaskDone(String[] inputWords, Task[] tasks) {
+        for (int i = 1; i < inputWords.length; i++) {
+            try {
+                int num = Integer.parseInt(inputWords[i]) - 1;
+                tasks[num].markAsDone();
+                System.out.println("Congrats! You've completed this task: [X] "
+                        + tasks[num].getTaskDesc());
+            } catch (NumberFormatException e) {
+                System.out.println(SOMETHING_WRONG +
+                        "Please ensure that you have entered the index of the task(s) to be marked as done");
+                break;
+            }
+        }
+    }
+
+    private static void listAllTasks(Task[] tasks) {
+        for (int i = 0; i < nextAdd; i++) {
+//            StringBuilder sb = new StringBuilder();
+            String done = tasks[i].isPending() ? "[ ]" : "[X]";
+            System.out.println((i + 1) + ": " + tasks[i].getType() + done + " "
+                    + tasks[i].getTaskDesc() + " " + tasks[i].getAdditionalInfo());
```

Comment:
It is good practice to split a good line into several lines to increase readability.